diff --git a/testdata/baselines/reference/fourslash/inlayHints/effectInlayHintsDisabledPassthrough.baseline b/testdata/baselines/reference/fourslash/inlayHints/effectInlayHintsDisabledPassthrough.baseline
new file mode 100644
index 0000000000..1ca6d77837
--- /dev/null
+++ b/testdata/baselines/reference/fourslash/inlayHints/effectInlayHintsDisabledPassthrough.baseline
@@ -0,0 +1,106 @@
+// === Inlay Hints ===
+export function standardShouldAppear() {
+                                      ^
+{
+  "position": {
+    "line": 2,
+    "character": 38
+  },
+  "label": [
+    {
+      "value": ": "
+    },
+    {
+      "value": "number"
+    }
+  ],
+  "kind": 1,
+  "paddingLeft": true
+}
+
+export const sample = Effect.gen(function*() {
+                                            ^
+{
+  "position": {
+    "line": 6,
+    "character": 44
+  },
+  "label": [
+    {
+      "value": ": "
+    },
+    {
+      "value": "Generator",
+      "location": {
+        "uri": "bundled:///libs/lib.es2015.generator.d.ts",
+        "range": {
+          "start": {
+            "line": 0,
+            "character": 0
+          },
+          "end": {
+            "line": 0,
+            "character": 0
+          }
+        }
+      }
+    },
+    {
+      "value": "<"
+    },
+    {
+      "value": "Effect",
+      "location": {
+        "uri": "file:///node_modules/effect/dist/Effect.d.ts",
+        "range": {
+          "start": {
+            "line": 132,
+            "character": 17
+          },
+          "end": {
+            "line": 132,
+            "character": 23
+          }
+        }
+      }
+    },
+    {
+      "value": "<"
+    },
+    {
+      "value": "never"
+    },
+    {
+      "value": ","
+    },
+    {
+      "value": "string"
+    },
+    {
+      "value": ","
+    },
+    {
+      "value": "never"
+    },
+    {
+      "value": ">"
+    },
+    {
+      "value": ","
+    },
+    {
+      "value": "number"
+    },
+    {
+      "value": ","
+    },
+    {
+      "value": "any"
+    },
+    {
+      "value": ">"
+    }
+  ],
+  "kind": 1,
+  "paddingLeft": true
+}
\ No newline at end of file
diff --git a/testdata/baselines/reference/fourslash/inlayHints/effectInlayHintsFnSuppression.baseline b/testdata/baselines/reference/fourslash/inlayHints/effectInlayHintsFnSuppression.baseline
new file mode 100644
index 0000000000..265c57ea76
--- /dev/null
+++ b/testdata/baselines/reference/fourslash/inlayHints/effectInlayHintsFnSuppression.baseline
@@ -0,0 +1,19 @@
+// === Inlay Hints ===
+export function standardShouldAppear() {
+                                      ^
+{
+  "position": {
+    "line": 2,
+    "character": 38
+  },
+  "label": [
+    {
+      "value": ": "
+    },
+    {
+      "value": "number"
+    }
+  ],
+  "kind": 1,
+  "paddingLeft": true
+}
\ No newline at end of file
diff --git a/testdata/baselines/reference/fourslash/inlayHints/effectInlayHintsFnUntracedSuppression.baseline b/testdata/baselines/reference/fourslash/inlayHints/effectInlayHintsFnUntracedSuppression.baseline
new file mode 100644
index 0000000000..265c57ea76
--- /dev/null
+++ b/testdata/baselines/reference/fourslash/inlayHints/effectInlayHintsFnUntracedSuppression.baseline
@@ -0,0 +1,19 @@
+// === Inlay Hints ===
+export function standardShouldAppear() {
+                                      ^
+{
+  "position": {
+    "line": 2,
+    "character": 38
+  },
+  "label": [
+    {
+      "value": ": "
+    },
+    {
+      "value": "number"
+    }
+  ],
+  "kind": 1,
+  "paddingLeft": true
+}
\ No newline at end of file
diff --git a/testdata/baselines/reference/fourslash/inlayHints/effectInlayHintsGenSuppression.baseline b/testdata/baselines/reference/fourslash/inlayHints/effectInlayHintsGenSuppression.baseline
new file mode 100644
index 0000000000..265c57ea76
--- /dev/null
+++ b/testdata/baselines/reference/fourslash/inlayHints/effectInlayHintsGenSuppression.baseline
@@ -0,0 +1,19 @@
+// === Inlay Hints ===
+export function standardShouldAppear() {
+                                      ^
+{
+  "position": {
+    "line": 2,
+    "character": 38
+  },
+  "label": [
+    {
+      "value": ": "
+    },
+    {
+      "value": "number"
+    }
+  ],
+  "kind": 1,
+  "paddingLeft": true
+}
\ No newline at end of file
diff --git a/testdata/baselines/reference/fourslash/inlayHints/effectInlayHintsNonEffectNotSuppressed.baseline b/testdata/baselines/reference/fourslash/inlayHints/effectInlayHintsNonEffectNotSuppressed.baseline
new file mode 100644
index 0000000000..c9a659ffbc
--- /dev/null
+++ b/testdata/baselines/reference/fourslash/inlayHints/effectInlayHintsNonEffectNotSuppressed.baseline
@@ -0,0 +1,19 @@
+// === Inlay Hints ===
+export function standardShouldAppear() {
+                                      ^
+{
+  "position": {
+    "line": 0,
+    "character": 38
+  },
+  "label": [
+    {
+      "value": ": "
+    },
+    {
+      "value": "number"
+    }
+  ],
+  "kind": 1,
+  "paddingLeft": true
+}
\ No newline at end of file
